codemod/spotlight_util.py

```diff
@@ -0,0 +1,34 @@
+#!/usr/bin/env python
+
+# Copyright (c) 2007-2008 Facebook
+
+from Cocoa import *
+
+
+class SpotlightUtil(object):
+    """
+    Searches a directory hierarchy using OS/X spotlight
+    """
+
+    @staticmethod
+    def walk_directory(root_directory, match_pattern):
+        query = NSMetadataQuery.alloc().init()
+        # going to trust that the user specified the proper
+        # spotlight format pattern
+        predicate = NSPredicate.predicateWithFormat_(
+            "(kMDItemTextContent = \"" + match_pattern + "\")")
+        if root_directory is not None:
+            query.setSearchScopes_([root_directory])
+        query.setPredicate_(predicate)
+        query.startQuery()
+        # TODO: figure out how to do this asynchronously if possible
+        while query.isGathering():
```

Comment:
So, the other way you can actually make this work is via NSNotificationCenter and 
`NSMetadataQueryDidFinishGatheringNotification`.

I am not good at PyObjC, but here is an ObjC implementation that would work:
```objc
- (NSArray *)resultsForSearchString:(NSString *)searchString inFolders:(NSSet *)paths
{
    NSMetadataQuery *query = [[NSMetadataQuery alloc] init];
    NSPredicate *search = [NSPredicate predicateWithFormat:searchString];
    [query setPredicate:search];
    if (paths) {
        NSMutableArray *pathURLs = [NSMutableArray array];
        for (NSString *path in paths) {
            NSURL *pathURL = [NSURL fileURLWithPath:path];
            [pathURLs addObject:pathURL];
        }
        [query setSearchScopes:pathURLs];
    }
    [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(_done:) name:NSMetadataQueryDidFinishGatheringNotification object:query];

    if (query.startQuery) {
        CFRunLoopRun();
    } else {
        NSLog(@"Failed to start a query");
    }
    [[NSNotificationCenter defaultCenter] removeObserver:self name:NSMetadataQueryDidFinishGatheringNotification object:nil];
    return query.results;
}

- (void)_done:(NSNotification *)notification {
    [notification.object stopQuery];
    CFRunLoopStop(CFRunLoopGetCurrent());
}
```